pkg/controller/configmap/config.go

```diff
@@ -324,12 +324,13 @@ func Render(configmap *corev1.ConfigMap, ncpReplicas int32, nsxSecret *corev1.Se
 		}
 	}
 	if haEnabled {
-		renderData.Data[operatortypes.NcpReplicasKey] = ncpReplicas
```

Comment:
I think the code could change to:

```
if not haEnabled and *ncpReplicas != 1 {
        log.Info(fmt.Sprintf("Set nsx-ncp deployment replicas to 1 instead of %d as HA is disabled",
	        *ncpReplicas))
        *ncpReplicas = 1
}
renderData.Data[operatortypes.NcpReplicasKey] = *ncpReplicas
```

Reply:
No, consider this case: haEnabled=false && ncpReplicas=3
In your above code, the renderData.Data[operatortypes.NcpReplicasKey]  will be 3, it's not expected.

Reply:
I don't understand why it will be 3 not 1?

Reply:
Sorry my bad. You are right. Will improve it in the next patch.